src/Workspaces/Core/Portable/Remote/RemoteHostSessions.cs

```diff
@@ -0,0 +1,280 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Execution;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.Remote
+{
+    /// <summary>
+    /// Helper type of common session case
+    /// 
+    /// RemoteHostClient.Session with 1 solution
+    /// </summary>
+    internal sealed class SessionWithSolution : IDisposable
```

Comment:
this is not session. this is something else. I am trying to make session as pure vs to remote host session that doesnt attached to any solution or cancellation. this is a thing for people who want a thing that want these 3 to be tied (solution, cancellation, session)